### PR TITLE
Fixes drawing on android

### PR DESCRIPTION
**Pull request by fundies**
GLES doesn't support texture2d and doesn't do bgra->rgba conversion so this pr provides fixes for both. Also, SDL behaves strangely when the window isn't fullscreen on android so I just force fullscreen for now.

Please merge https://github.com/enigma-dev/enigma-android/pull/1 & #1899 before this

### CODE REVIEW

**Comment by fundies** [comment]
android

**Comment by RobertBColton** [comment]
Are you using GLAD or what? I remember what I think we did to solve the BGRA problem before, we used the extension which was supported on like 98% of hardware according to some Khronos statistics.
https://www.khronos.org/registry/OpenGL/extensions/EXT/EXT_texture_format_BGRA8888.txt

File: ENIGMAsystem/SHELL/Graphics_Systems/OpenGL-Common/textures.cpp [hunk 1]
```diff
@@ -29,6 +29,22 @@
...
+      out[offset] = in[offset + 2];
+      out[offset + 1] = in[offset + 1];
+      out[offset + 2] = in[offset];
+      out[offset + 3] = in[offset + 3];
+      offset += 4;
+    }
+  }
+  return out;
+}
+
```
Comment (RobertBColton) [inline ENIGMAsystem/SHELL/Graphics_Systems/OpenGL-Common/textures.cpp]: Are you using GLAD or what? I remember what I think we did to solve the BGRA problem before, we used the extension which was supported on like 98% of hardware according to some Khronos statistics.
https://www.khronos.org/registry/OpenGL/extensions/EXT/EXT_texture_format_BGRA8888.txt

Edit: Correction* it's supported by 99.2% of devices.
https://opengles.gpuinfo.org/listextensions.php

BGRA GLES EXT Support

Comment (fundies) [reply ENIGMAsystem/SHELL/Graphics_Systems/OpenGL-Common/textures.cpp]: If i put bgra & bgra everything blue and red are swapped even on desktop. The real solution is to just kill this bgra nonsense in the engine but thats far too much work for me. Right now you have opengl doing the conversion but gles cannot do that conversion even if it supports bgra.

Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Graphics_Systems/OpenGL-Common/textures.cpp]: No I mean put it back the way that it was (BGRA & **_RGBA_**) and add the extension. The desktop was obviously fine before you changed _anything_! I recall API tracing GMS and seeing that it was in fact still using BGRA internally and I also recall finding somewhere that it too used this extension.

Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Graphics_Systems/OpenGL-Common/textures.cpp]: See GameMaker discussion as well:
https://forum.yoyogames.com/index.php?threads/switching-gms2-from-bgr-to-standard-rgb.33526/

Comment (fundies) [reply ENIGMAsystem/SHELL/Graphics_Systems/OpenGL-Common/textures.cpp]: The extension has nothing todo w/ conversion. bgra is already enable but GLES will not convert bgra to rgba regardless of any extension

Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Graphics_Systems/OpenGL-Common/textures.cpp]: Try GL_BGRA_EXT

Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Graphics_Systems/OpenGL-Common/textures.cpp]: Also TKG made a helper that appears to do the same thing:
https://github.com/enigma-dev/enigma-dev/pull/1904/files#diff-1246ecb7fd9dc46a9b17aa578b0d5c0bR37

Comment (JoshDreamland) [reply ENIGMAsystem/SHELL/Graphics_Systems/OpenGL-Common/textures.cpp]: I think Sam just checked this method in as part of his icon PR. Please try to reuse that.

Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Graphics_Systems/OpenGL-Common/textures.cpp]: fundies is also being difficult and I don't know whether to believe him about BGRA.

**Comment by RobertBColton** [comment]
Who's Travis?

**Comment by fundies** [comment]
> Why's Travis failing?

Because I'm calling delete wrong somehow. I dunno what handles deletion of pxdata after uploading to gl. I haven't dug too deep yet

File: ENIGMAsystem/SHELL/Graphics_Systems/OpenGL-Common/textures.cpp (hunk 1 above)
Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Graphics_Systems/OpenGL-Common/textures.cpp:47]: Here's that thing tkg added.
https://github.com/enigma-dev/enigma-dev/blob/2d1220f23fa6dd057bf478eba8f25e5a82d34b6d/ENIGMAsystem/SHELL/Universal_System/image_formats.cpp#L188

**Pull request closed without merging**